documentation/modules/ROOT/pages/02-architecture.adoc

```diff
@@ -1,104 +1,255 @@
 ====
 
-[IMPORTANT]
-====
-This is IMPORTANT
-====
+=== _Work Ledger_ analogy
+
+A good way of thinking about the _Event Mesh_ and its persistent queue backend is the _Work Ledger_ analogy.
+Like in olden days, the clerk was keeping his to-do work in the _Work Ledger_ (e.g. a tray for paper forms).
+Then he was picking the next form, and processing it, making changes within the physical file cabinets.
+In case of rare and unexpected issues (e.g. Invoicing dept being unavailable), the clerk would just put the data back onto the _Work Ledger_ to be processed later.
+
+The _Event Mesh_ is processing the data in very similar fashion.
+The data is held in the _Event Mesh_ only until the system successfully consumes it.
+
+=== Differences from the _Event Sourcing_
+
+The _Event Mesh_ pattern could be mistaken for _Event Sourcing_, as both are Event-Driven approaches (_EDA_) to application architecture.
+However, _Event Mesh_ has few improvements over the shortcomings of _Event Sourcing_ approach.
 
+The data is held in the _Event Mesh_ only until the system successfully consumes it, settling the data in various datastores to a consistent state.
+This effectively avoids the need to keep the applications backward compatible with all the events ever emitted.
+Introducing breaking changes in the event schema is as easy as making sure to consume all the events of given type from the _Event Mesh_.
+This also works for the systems which can't have downtime windows.
+The applications could have short-lived backward compatible layers for such situations.
+When all the events are processed, the backward compatible code may be removed simplifying the maintenance.
 
-=== Creating  tables
- 
-[cols="1a,1a,1a"]
+Because in the long-term, the regular datastores are the source of truth for the system, all traditional techniques for application maintenance apply well.
+It is also, easier to understand for developers, as it avoids sophisticated event handlers logic, and reconciling into the read database abstraction.
+
+=== Differences from the _Service Mesh_
+
+Worth pointing out are the differences from the _Service Mesh_ pattern.
+The _Service Mesh_ pattern is intended to improve the resilience of synchronous communications which return the response.
+The _Service Mesh_ effectively raises the uptime of the dependent endpoints by retrying and backoff policies.
+The uptime can't be raised to 100%, though, so _Service Mesh_ still can lose the messages.
+
+The table below captures the key differences:
+
+[cols=".^1,.^2,.^2"]
 |===
-| *Column A*  | Column *A* | _Column C_
-|
-* Lorem Ipsum
-* Lorem Ipsum
+|| Service Mesh | Event Mesh
 
 |
-* Lorem Ipsum
-* Lorem Ipsum
+*Similarities*
+
+2+a|
+
+* Flexibility
+* Robustness
+* Decoupling
 
 |
-* Lorem Ipsum
-* Lorem Ipsum
+*Differences*
+a|
+* Synchronous
+* Request and response
+* Better for queries
+
+a|
+* Asynchronous
+* Events
+* Better for commands
+
 |===
 
+=== Supporting Legacy Systems
```

Comment:
CloudEvent is key here - in binary mode, the exta metadata is just Headers (true for kafka/http) and actual payload stay intact .

Reply:
I don't follow your comment here. Can you re-state what are you proposing?